src/main/java/ui/components/pickers/LabelPickerUILogic.java

```diff
@@ -8,6 +8,28 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+/**
+ * Handles the logic of the label picker UI.
+ *
+ * The crux of the logic is as follows:
+ * 1. When the text in the LabelPicker's text field changes, processTextFieldChange is called.
+ * 2. If the text is not empty, this method will try to determine if the the user typed in a character or backspaced.
+ *   2a. Typing is handled by addCharHandler
+ *     i. If the user typed ' ', it will attempt to add the current highlighted label (if there is).
+ *     ii. Else it is an ongoing query
```

Comment:
You may want to explain somewhere what a `query` is. That's the sort of thing a comment should explain because the code itself cannot tell the read what that term means.